src/main/kotlin/com/delta/network/GameController.kt

```diff
@@ -51,20 +51,36 @@ class GameController(
      * Вызывается внутри [listener], когда ему приходит сигнал от сервера.
      */
     private fun updateGameState(gameState: Tilous) {
-        // TODO()
-        // 1. Обновите состояние игры (this.gameState.game)
 
-        // 2. Если игра еще не началась И попытка узнать свой id успешна
-        //      -- поменять фазу на ожидание остальных игроков
+        // 1. Update the game state
+        this.gameState.game = gameState
+        // Possible logic error. If the game state is
+        // 2. If the game has not started AND the attempt to find out the ID is successful
+        if (this.gameState.phase == GamePhase.NOT_STARTED && this.gameState.playerID != null) {
```

Comment:
This will through null pointer exception if this.gameState.playerID is null. I would put this code into if statement which checks that playerID is not null.